pageserver/src/waldecoder.rs

```diff
@@ -368,6 +368,8 @@ impl XlHeapMultiInsert {
 pub struct XlHeapDelete {
     pub xmax: TransactionId,
     pub offnum: OffsetNumber,
+    pub _padding: u16,
```

Comment:
May be delete `#[repr(C)]` here? It is kind of misleading -- we don't do any transmute to this struct.

Reply:
Yeah. I'm not very happy with these structs and the ad hoc WAL decoding we do here in general. These structs really should be generated from the postgres headers with bindgen, and should live in postgres_ffi, since they're specific to the PostgreSQL major version. Much of the code in decode_wal_record, too, I think.

Reply:
Re-opening https://github.com/zenithdb/zenith/issues/220 to track that.